client/src/components/NationwideMap.vue

```diff
@@ -281,63 +307,68 @@ export default defineComponent({
 
-      // Check whether there's a layer selected in the router.
-      this.state.setCurrentDataLayer(
-        this.state.dataLayers.find(
-          (layer) => layer.styleLayer.id == router.currentRoute.value.query?.layer,
-        ) ?? leadServiceLinesByWaterSystemLayer,
-      );
+      // If the map has nothing on it, check the current data layer.
+      if (this.visibleLayer == null && this.currentDataLayerId != null) {
+        this.updateMapOnDataLayerChange(ALL_DATA_LAYERS.get(this.currentDataLayerId));
+      }
+
     },
 
     /**
      * Creates base Mapbox map and configures interactions and styles for
      * layers.
      */
     async createMap(): Promise<void> {
-      try {
-        this.map = new mapbox.Map({
-          // Removes watermark by Mapbox.
-          attributionControl: false,
-          center: this.center,
-          container: 'map-container',
-          style: 'mapbox://styles/blueconduit/cku6hkwe72uzz19s75j1lxw3x?optimize=true',
-          zoom: 4,
-        });
+      const zoom = 4;
+      this.map = new mapbox.Map({
+        // Removes watermark by Mapbox.
+        attributionControl: false,
+        center: this.center,
+        container: 'map-container',
+        style: 'mapbox://styles/blueconduit/cku6hkwe72uzz19s75j1lxw3x?optimize=true',
+        zoom: zoom,
+      });
 
-        this.map.on('load', this.configureMap);
-        this.map.scrollZoom.disable();
-      } catch (err) {
-        // TODO: Add error handling.
-        console.log('Error: ', err);
-      }
+      this.map.on('load', this.configureMap);
+      this.map.on('error', (error) => {
+        console.log(`Error loading tiles: ${error.error} `);
+        console.log(error.error.stack);
+      });
+
+      this.map.scrollZoom.disable();
+      dispatch(setZoom(zoom));
     },
   },
   mounted() {
     this.createMap();
   },
   watch: {
-    // Listens to app state to toggle layers.
-    'state.currentDataLayer': function(newDataLayer: DataLayer, oldDataLayer: DataLayer) {
-      this.updateMapOnDataLayerChange(newDataLayer, oldDataLayer);
-    },
-    // Listens to query param to toggle layers.
-    routerLayer: function(newLayer: string) {
-      if (newLayer != null) {
-        this.setDataLayerVisibility(newLayer, true);
-      }
+    // Listens to map state to toggle different layers.
+    'mapState.mapData.currentDataLayerId': {
+      handler: function(newDataLayerId: MapLayer) {
+        if (newDataLayerId == null) {
+          return;
+        }
+        this.updateMapOnDataLayerChange(ALL_DATA_LAYERS.get(newDataLayerId));
+      },
     },
     // Listen for changes to lat/long to update map location.
     'geoState.geoids': function() {
       const lat = this.geoState?.geoids?.lat;
       const long = this.geoState?.geoids?.long;
 
       if (lat != null && long != null) {
-        const marker = new mapboxgl.Marker({
+        // Remove the old marker.
+        if (this.marker != null) {
+          this.marker.remove();
```

Comment:
the only change i made was moving this outside of the if

Reply:
oo good point - swapped it